Terminal.Gui/Views/DateField.cs

```diff
@@ -121,13 +119,13 @@ void DateField_Changing (object sender, TextChangingEventArgs e)
 					break;
 				}
 			}
-			spaces += _fieldLen;
+			spaces += FormatLength;
 			string trimedText = e.NewText [..spaces];
-			spaces -= _fieldLen;
+			spaces -= FormatLength;
 			trimedText = trimedText.Replace (new string (' ', spaces), " ");
-			var date = Convert.ToDateTime (trimedText, ccFmt).ToString (ccFmt.ShortDatePattern);
+			var date = Convert.ToDateTime (trimedText).ToString (_format.Trim ());
 			if ($" {date}" != e.NewText) {
-				e.NewText = $" {date}";
+				e.NewText = $" {date}".Replace ("\u200f", "");
```

Comment:
Please make "\u200f" a const or at least document what the heck it is. Code written for this library should assume people reading it are 6-year-olds. ;-)

Reply:
Good idea, done